src/mam4xx/ndrop.hpp

```diff
@@ -130,6 +245,1506 @@ void maxsat(
+    eta[imode] = etafactor1 * etafactor2[imode];
+  } // end imode
+
+  Real supersat = zero;
+
+  maxsat(zeta, eta, nmode, ssat_crit_imode, supersat);
+  // ([fraction]))
+  const Real lnsupersat = haero::log(supersat);
+
+  // Use maximum supersaturation to calculate aerosol activation output
+  for (int imode = 0; imode < nmode; ++imode) {
+    // [unitless]
+    const Real arg_erf_n =
+        twothird * (lnsm[imode] - lnsupersat) / (sq2 * alogsig[imode]);
+
+    fn[imode] = half * (one - haero::erf(arg_erf_n)); // activated number
+
+    const Real arg_erf_m = arg_erf_n - 1.5 * sq2 * alogsig[imode];
+    fm[imode] = half * (one - haero::erf(arg_erf_m)); // activated mass
+    fluxn[imode] = fn[imode] * w_in; // activated aerosol number flux
+    fluxm[imode] = fm[imode] * w_in; // activated aerosol mass flux
+  }
+  // NOTE: what is this??
+  // is vertical velocity equal to flux of activated aerosol fraction assuming
+  // 100% activation [m/s]?
+  flux_fullact = w_in;
+
+} // activate_modal
+
+KOKKOS_INLINE_FUNCTION
+void get_activate_frac(
+    const Real state_q_kload[nvars], const Real air_density_kload,
+    const Real air_density_kk, const Real wtke,
+    const Real tair, // in
+    const int lspectype_amode[maxd_aspectype][AeroConfig::num_modes()],
+    const Real specdens_amode[maxd_aspectype],
+    const Real spechygro[maxd_aspectype],
+    const int lmassptr_amode[maxd_aspectype][AeroConfig::num_modes()],
+    const Real voltonumbhi_amode[AeroConfig::num_modes()],
+    const Real voltonumblo_amode[AeroConfig::num_modes()],
+    const int numptr_amode[AeroConfig::num_modes()],
+    const int nspec_amode[maxd_aspectype],
+    const Real exp45logsig[AeroConfig::num_modes()],
+    const Real alogsig[AeroConfig::num_modes()], const Real aten,
+    Real fn[AeroConfig::num_modes()], Real fm[AeroConfig::num_modes()],
+    Real fluxn[AeroConfig::num_modes()], Real fluxm[AeroConfig::num_modes()],
+    Real &flux_fullact) {
+
+  // input arguments
+  //  @param [in] state_q_kload(:)         aerosol mmrs at level from which to
+  //  load aerosol [kg/kg]
+  //  @param [in] cs_kload     air density at level from which to load aerosol
+  //  [kg/m3]
+  //  @param [in] cs_kk        air density at actual vertical level [kg/m3]
+  //  @param [in] wtke        subgrid vertical velocity [m/s]
+  //  @param [in]  tair        air temperature [K]
+
+  // output arguments
+  // @param [out]  fn(:)        number fraction of aerosols activated
+  // [fraction]
+  // @param [out]   fm(:)        mass fraction of aerosols activated
+  // [fraction]
+  // @param [out]   fluxn(:)     flux of activated aerosol number fraction
+  // into cloud [m/s]
+  // @param [out]   fluxm(:)     flux of activated aerosol mass fraction into
+  // cloud [m/s]
+  // @param [out]   flux_fullact flux of activated aerosol fraction assuming
+  // 100% activation [m/s]
+  const Real zero = 0;
+  const int nmodes = AeroConfig::num_modes();
```

Comment:
Should this (and statements like it) be `constexpr int` instead of `const int` to make sure that it can be used to provide a size for automatic arrays?